### DISCUSSION TITLE
Triggered function on internal auth.identities not working

### DISCUSSION THREAD

**Discussion by tzvc** [supabase/supabase#19777] [Questions] [answered]
Hey there,

I'm trying to create a triggered function that ads the name and avatar_url field from supabase's internal auth.identities table to the users public.profiles row on new row insert/update.

After creating the function and the trigger, nothing happens. New users using Google Provider dont see their name and avatar_url updated. I've check the postgres logs, I see no errors.

What am I missing?

Here's my code:

```SQL
CREATE OR REPLACE FUNCTION update_profiles_on_identity_change()
RETURNS TRIGGER AS $$
BEGIN
    -- Extract name and avatar_url from identity_data
    UPDATE public.profiles
    SET name = NEW.identity_data ->> 'name',
        avatar_url = NEW.identity_data ->> 'avatar_url'
    WHERE id = NEW.user_id;

    RETURN NEW;
END;
$$ LANGUAGE plpgsql;

-- Create the trigger for both INSERT and UPDATE
CREATE OR REPLACE TRIGGER trigger_update_profiles_from_identity
AFTER INSERT OR UPDATE ON auth.identities
FOR EACH ROW
EXECUTE FUNCTION update_profiles_on_identity_change();
```

**Comment by GaryAustin1** [comment, marked as answer]
Are you getting errors in the Postgres logs?

I would think the function needs to be security definer type though. Normally the auth role does not have access to the public schema. 

Not commenting on if your json formatting or if the data is even there as I’ve not seen anyone using the identities  table for triggers and don’t use OAuth myself.

**Comment by tzvc** [comment]
Thanks @GaryAustin1 `security definer` did the job to override RLS.

for reference here's the updated code:

```sql
CREATE OR REPLACE FUNCTION update_profiles_on_identity_change()
RETURNS TRIGGER AS $$
BEGIN
    -- Extract name and avatar_url from identity_data
    UPDATE public.profiles
    SET name = NEW.identity_data ->> 'name',
        avatar_url = NEW.identity_data ->> 'avatar_url'
    WHERE id = NEW.user_id;

    RETURN NEW;
END;
$$ LANGUAGE plpgsql SECURITY DEFINER;

-- Create the trigger for both INSERT and UPDATE
CREATE OR REPLACE TRIGGER trigger_update_profiles_from_identity
AFTER INSERT OR UPDATE ON auth.identities
FOR EACH ROW
EXECUTE FUNCTION update_profiles_on_identity_change();
```
